core-library/src/main/java/org/silverpeas/core/index/search/SimpleSearchEngine.java

```diff
@@ -108,4 +120,100 @@ private boolean isSpellingNeeded(List<MatchingIndexEntry> results) {
     SearchCompletion completion = new SearchCompletion();
     return completion.getSuggestions(keywordFragment);
   }
+
+  private List<MatchingIndexEntry> filterMatchingIndexEntries(
+      List<MatchingIndexEntry> matchingIndexEntries, String userId) {
+    if (matchingIndexEntries == null || matchingIndexEntries.isEmpty()) {
+      return new ArrayList<>();
+    }
+    List<MatchingIndexEntry> results = new ArrayList<>(matchingIndexEntries.size());
+    ComponentAuthorization authorization = null;
+    try {
+      authorization = getSecurityIntf();
+      authorization.enableCache();
+    } catch (Exception e) {
+      SilverLogger.getLogger(this).error(e.getMessage(), e);
+    }
+
+    for (MatchingIndexEntry result : matchingIndexEntries) {
+      if (!isMatchingIndexEntryAvailable(result, userId, authorization)) {
+        continue;
+      }
+      results.add(result);
+    }
+
+    authorization.disableCache();
+
+    return results;
+  }
+
+  private boolean isMatchingIndexEntryAvailable(MatchingIndexEntry mie, String userId,
+      ComponentAuthorization authorization) {
+    // Do not filter and check external components
+    if (enableExternalSearch && isExternalComponent(mie.getServerName())) {
+      mie.setExternalResult(true);
+      // Filter only Publication and Node data
+      String objectType = mie.getObjectType();
+      if ("Versioning".equals(objectType) || "Publication".equals(objectType)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this if-then-else statement by a single return statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1126)